ast.ts

```diff
@@ -49,5 +52,9 @@ export enum BinOp { Plus, Minus, Mul, IDiv, Mod, Eq, Neq, Lte, Gte, Lt, Gt, Is,
 export enum UniOp { Neg, Not };
 
 export type Value =
-    Literal
-  | { tag: "object", name: string, address: number}
+    //Literal
+    { tag: "num", value: BigInt }
+  | { tag: "bool", value: boolean }
+  | { tag: "string", value: number}
```

Comment:
There's an interesting inconsistency here where `BigInt` is expected to convert its result (even if heap allocated) but strings aren't. Let's use `Literal` again here, and figure out how to return a string. If you need a hack to get started, you can always return strings as `class`-tagged values to start and later do the conversion.

Reply:
Done. I have changed it to Literal.